tests/test_stemma_NJ.py

```diff
@@ -0,0 +1,103 @@
+                                                  5.,  4.,  2.,  4.,  6.,  6.],
+                                              [15., 10., 14., 14.,  0., 13., 13.,
+                                                  12., 10., 12., 10., 12., 12.],
+                                              [2.,  3.,  7.,  7., 13.,  0.,  0.,
+                                                  2.,  3.,  5.,  3.,  2.,  2.],
+                                              [2.,  3.,  7.,  7., 13.,  0.,  0.,
+                                                  2.,  3.,  5.,  3.,  2.,  2.],
+                                              [3.,  2.,  5.,  5., 12.,  2.,  2.,
+                                                  0.,  2.,  4.,  2.,  3.,  3.],
+                                              [5.,  0.,  4.,  4., 10.,  3.,  3.,
+                                                  2.,  0.,  2.,  0.,  2.,  2.],
+                                              [7.,  2.,  2.,  2., 12.,  5.,  5.,
+                                                  4.,  2.,  0.,  2.,  4.,  4.],
+                                              [5.,  0.,  4.,  4., 10.,  3.,  3.,
+                                                  2.,  0.,  2.,  0.,  2.,  2.],
+                                              [4.,  2.,  6.,  6., 12.,  2.,  2.,
+                                                  3.,  2.,  4.,  2.,  0.,  0.],
+                                              [4.,  2.,  6.,  6., 12.,  2.,  2.,  3.,  2.,  4.,  2.,  0.,  0.]])
+        self.testing_edge_dict = {'N_1,c': 11.0,
+                                  'N_1,d': 17.0,
+                                  'N_2,a': 4.75,
+                                  'N_2,N_1': 7.25,
+                                  'N_3,b': 6.75,
+                                  'N_3,e': 14.25,
+                                  'N_2,N_3': 4.75}
+
+    @staticmethod
+    def distance_test(text1: str, text2: str) -> float:
+        """Function used to test distance parameter.
+        """
+        return abs(len(text1) - len(text2))
+
+    def test_getters(self):
+        """Testing getters for class properties."""
+        self.stemmaNJ = StemmaNJ()
+        self.stemmaNJ.compute(
+            folder_path=self.stemma_folder_path, distance=levenshtein)
+        self.assertTrue((self.stemmaNJ.dist_matrix.round(7) == self.test_distance_matrix.round(
+            7)).all(), msg="The distance matrix is not correct.")
+        self.assertEqual(self.stemmaNJ.distance, levenshtein,
+                         msg="The returned distance is not correct.")
+
+    def test_build_edges(self):
+        """Tests the _build_edges method."""
+        temp_stem = StemmaNJ()
+        temp_stem._dist_matrix = self.dist_mat_edge_test
+        temp_stem._manuscripts = {
+            "a": "", "b": "", "c": "", "d": "", "e": "", }
+        distance_dict, edge_list = temp_stem._build_edges()
+        self.assertDictEqual(distance_dict, self.testing_edge_dict,
```

Comment:
if this is testing differences between distance matrix, perhaps you can rename `self.testing_edge_dict` to something clearer ?

Reply:
Renamed it to edge_dictionary_reference